### PR TITLE
Add the ability to invoke effectors.

### PR DESCRIPTION
**Pull request by geomacy**
For a given effector, invoke it with

brooklyn-cli effectors APPID ENTITYID EFFECTORNAME [ params ]

where params is a sequence of values for the parameters of the effector, if any.

For example see the following sequence ("brooklyn-cli" has been aliased to "bk"):

$ bk applications
Id   Name   Status   Location
$
$
$ bk create myweb.yaml
response Status: 201 Created
...
$ bk applications
Id         Name             Status    Location
DRGUqnXs   My Web Cluster   RUNNING   fm7gDUks
$
$ bk entities DRGUqnXs
Id         Name     Type
dhwkYGGX   My Web   org.apache.brooklyn.entity.webapp.ControlledDynamicWebAppCluster
TTHOUkbx   My DB    org.apache.brooklyn.entity.database.mysql.MySqlNode
$ bk sensors DRGUqnXs dhwkYGGX | grep webapp.deployedWars
$
webapp.deployedWars                                    Names of archives/contexts that are currently deployed
$ bk effectors DRGUqnXs dhwkYGGX | grep deploy
deploy           Deploys the given artifact, from a source URL, to a given deployment filename/context                                                                                                  url,targetName
...
$ bk effectors DRGUqnXs dhwkYGGX deploy https://tomcat.apache.org/tomcat-6.0-doc/appdev/sample/sample.war mysample
response Status: 202 Accepted
response Headers: map[Content-Length:[0] Set-Cookie:[JSESSIONID_BROOKLYNJLKc8Z=cwjdngii3o919cawdrezzajb;Path=/] Content-Type:[application/json] Vary:[Accept-Encoding] Pragma:[no-cache] Date:[Thu, 26 Nov 2015 22:22:22 GMT] Expires:[Thu, 01 Jan 1970 00:00:00 GMT 0] Cache-Control:[no-cache, no-store] Server:[Jetty(9.2.13.v20150730)]]
response Body:
$
$ bk sensors DRGUqnXs dhwkYGGX | grep webapp.deployedWars
webapp.deployedWars                                    Names of archives/contexts that are currently deployed                                                           ["/mysample"]
bk effectors DRGUqnXs DRGUqnXs stop
response Status: 202 Accepted
response Headers: map[Server:[Jetty(9.2.13.v20150730)] Set-Cookie:[JSESSIONID_BROOKLYNJLKc8Z=6ker7g8hfzmn1gj07i3znbq3l;Path=/] Expires:[Thu, 01 Jan 1970 00:00:00 GMT 0] Vary:[Accept-Encoding] Content-Length:[0] Date:[Thu, 26 Nov 2015 22:27:31 GMT] Content-Type:[application/json] Cache-Control:[no-cache, no-store] Pragma:[no-cache]]
response Body:
$
$ bk applications
Id   Name   Status   Location
$

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 brooklyn-cli
+.idea
+*.iml
+temp_test.go
```
Comment (geomacy) [inline .gitignore:4]: Note - the addition of temp_test.go is deliberate.   This is to allow one to have the file in question there for quick test purposes.

File: commands/effectors.go [hunk 2]
```diff
@@ -40,3 +55,48 @@ func (cmd *Effectors) Run(c *cli.Context) {
...
+	effectorArgs := c.Args()[3:]
+
+	// get the effector
+	effectors := entity_effectors.EffectorList(cmd.network, c.Args()[0], c.Args()[1])
+	effector, err := findEffector(effectorName, effectors)
+	if nil != err {
+		return err
+	}
+
+	// get its parameters and check we have an arg for each
```
Comment (robertgmoss) [inline commands/effectors.go:72]: Some parameters for some effectors are optional

**Comment by robertgmoss** [comment]
Please rebase or close

**Comment by geomacy** [comment]
Closing this,  will probably do it somewhat differently.

**Pull request closed without merging**